server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/PersistComponentsStep.java

```diff
@@ -112,6 +114,15 @@ private void disableRemainingComponents(DbSession dbSession, Collection<Componen
     disabledComponentsHolder.setUuids(uuids);
   }
 
+  private void ensureConsistentVisibility(DbSession dbSession, String projectUuid, boolean isRootPrivate) {
+    dbClient.componentDao().setPrivateForRootComponentUuid(dbSession, projectUuid, isRootPrivate);
+  }
+
+  private boolean isRootPrivate(String rootKey, Map<String, ComponentDto> existingDtosByKeys) {
+    ComponentDto projectDto = existingDtosByKeys.get(rootKey);
+    return projectDto == null ? false /*FIXME actually use default configured for current organization*/ : projectDto.isPrivate();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134)